schema/python/schemaorgutils/serializer.py

```diff
@@ -0,0 +1,357 @@
+            date: Python time object of protobuf time.
+        """
+
+        hours = getattr(obj, 'hours')
+        minutes = getattr(obj, 'minutes')
+        seconds = getattr(obj, 'seconds')
+        timezone = getattr(obj, 'timezone')
+
+        time = datetime.time(hour=hours, minute=minutes, second=seconds)
+        
+        if timezone:
+            time_string = time.isoformat()
+            time_string = time_string + timezone
+            time = datetime.time.fromisoformat(time_string)
+        
+        return time.isoformat()
+
+    def __datetime_to_string(self, obj):
+        """Convert a protobuf datetime object to isostring format.
+
+        Args:
+            obj (protobuf object): Protobuf object of datetime.
+            schema (module): Module containing compiled proto schema.
+
+        Returns:
+            date_time: Isostring format of proto datetime.
+        """
+
+        year = getattr(getattr(obj, 'date'), 'year')
+        month = getattr(getattr(obj, 'date'), 'month')
+        day = getattr(getattr(obj, 'date'), 'day')
+        hours = getattr(getattr(obj, 'time'), 'hours')
+        minutes = getattr(getattr(obj, 'time'), 'minutes')
+        seconds = getattr(getattr(obj, 'time'), 'seconds')
+        timezone = getattr(getattr(obj, 'time'), 'timezone')
+
+        date_time = datetime.datetime(
+            year=year,
+            month=month,
+            day=day,
+            hour=hours,
+            minute=minutes,
+            second=seconds)
+        
+        if timezone:
+            date_time_string = date_time.isoformat()
+            date_time_string = date_time_string + timezone
+            date_time = datetime.datetime.fromisoformat(date_time_string)
+
+        return date_time.isoformat()
+
+    def __duration_to_string(self, obj):
+        """Convert a proto duration object to ISO8601 string.
+
+        Args:
+            obj (protobuf object): Protobuf object of duration.
+
+        Returns:
+            string: The duration as ISO8601 string.
+        """
+        
+        sec = obj.seconds
+        d = datetime.timedelta(seconds=sec)
+        return isodate.duration_isoformat(d)
+    
+    def __quantitative_to_string(self, obj):
+        """Convert a proto quantitative object to string.
+
+        Args:
+            obj (protobuf object): Protobuf object of quantitative object.
+
+        Returns:
+            string: The quantitative object as string.
+        """
+        
+        value = obj.value
+        unit = obj.unit
+        return str(value) + " " + unit
+
+    def __check_primtive(self, obj):
+        """Check if an object is python primitive type.
+
+        Args:
+            obj (object): Object that needs to checked.
+
+        Returns:
+            bool: If it is primitive type or not.
+        """
+
+        for x in self._primitive_types:
+            if isinstance(obj, x):
+                return True
+        return False
+
+    def proto_to_dict(self, obj, schema):
```

Comment:
This method name is wrong. Instead it should be serialize_proto.
Similarly for other methods like serialize_enum, serialize_property, serialize_class